### PR TITLE
Add python 3.8/3.9 support for Windows GPU and Linux ARM64

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
1. Add python 3.8/3.9 support for Windows GPU and Linux ARM64
2. Delete jemalloc from cgmanifest.json.
3. Add onnx node test to Nuphar pipeline.
4. Change  $ANDROID_HOME/ndk-bundle  to $ANDROID_NDK_HOME. The later one is more accurate.
5. Delete Java GPU packaging pipeline
6. Remove test data download step in Nuget Mac OS pipeline. Because these machines are out of control and out of our network, it's hard to make it reliable and the data secure.
7. Fix a doc problem in c-api-artifacts-package-and-publish-steps-windows.yml. It shouldn't copy C_API.md, because the file has been moved into a different branch.
8. Delete the CI build docker file for Ubuntu cuda 9.x and Ubuntu x86 32 bits

9. And, due to some internal restrictions, I need to rename some of the agent pools

| Org | Old name  | New Name  |  
|---|---|---|
|onnxruntime | Linux-CPU | Linux-CPU-2019|
|aiinfra  |  Win-CPU-2019 |  Win-CPU-2021|

**Motivation and Context**
- Why is this change required? What problem does it solve?
To have full support for python 3.8/3.9.

Note: there is a tiny issue for the GPU java packaging part.  The package name is wrong. I'll address it separately.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 1]
```diff
@@ -802,8 +807,8 @@ ::std::vector<::std::basic_string<ORTCHAR_T>> GetParameterStrings() {
 #endif
 #endif
 
-// TENSORRT has too many test failures in the single node tests
-#if !defined(_WIN32) && !defined(USE_TENSORRT)
+// TENSORRT/TRT/Nuphar has too many test failures in the single node tests
```
Comment (edgchen1) [inline onnxruntime/test/providers/cpu/model_tests.cc:810]: mention OpenVINO in the comment instead?

File: tools/ci_build/github/windows/jar_gpu_packaging.ps1 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+$ErrorActionPreference = "Stop"
+Write-Output "Start"
+dir
+pushd onnxruntime-java-linux-x64
+Write-Output "Run 7z"
+7z a $Env:BUILD_BINARIESDIRECTORY\java-artifact\onnxruntime-java-win-x64\testing.jar libcustom_op_library.so
+Remove-Item -Path libcustom_op_library.so
+7z a $Env:BUILD_BINARIESDIRECTORY\java-artifact\onnxruntime-java-win-x64\onnxruntime-$Env:ONNXRUNTIMEVERSION.jar .
+popd
+7z a $Env:BUILD_BINARIESDIRECTORY\java-artifact\onnxruntime-java-win-x64\onnxruntime-$Env:ONNXRUNTIMEVERSION.jar .
```
Comment (edgchen1) [inline tools/ci_build/github/windows/jar_gpu_packaging.ps1:10]: duplicated lines?

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh [hunk 3]
```diff
@@ -45,23 +48,23 @@ if [ $RunTestNative = "true" ]; then
 
   inc="-I build/native/include"
 
-  if [ $IsMacOS = "true" ]; then
+  if [[ $IsMacOS == "True" || $IsMacOS == "true" ]]; then
```
Comment (edgchen1) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:51]: looks like there is a way to convert to lower case: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion (search for ${parameter,,pattern})
how about `${IsMacOs,,} == "true"`?

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh [hunk 4]
```diff
@@ -3,7 +3,6 @@
 # Licensed under the MIT License.
 
 LocalNuGetRepo=$1
-SourceRoot=$2
```
Comment (edgchen1) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: why not keep this parameter and pass the correct value? seems better to not depend on azure devops variable names like BUILD_SOURCESDIRECTORY outside of build definitions.

Comment (edgchen1) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: could add an additional parameter for the binaries directory

Comment (snnn) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: Yes, then I need to add another parameter, which requires modify more files(all the callers of this script). That's why I choose this.

Comment (snnn) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: Indeed I would prefer to pass less parameters when possible. The more we have, the more likely we could make some errors during the passing, for example, missing one parameter or the order is wrong.

**Review by edgchen1 — APPROVED** [review]

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh (hunk 4 above)
Comment (edgchen1) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: that's what error handling is for :)

Comment (snnn) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:6]: Oh, I forgot to mention, sometimes the file is called from docker, sometimes it runs directly in the host environment.

**Pull request merged**